### PR TITLE
doc: add more remarks regarding detached

### PR DESCRIPTION
**Pull request by anonrig**
Added an example for detecting detached state, and explained the current status of `detach` in browsers and in runtimes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,6 +111,24 @@ Owing to [messy history](https://esdiscuss.org/topic/arraybuffer-neutering) of T
 
 The `detached` getter is added to authoritatively determine whether an `ArrayBuffer` is detached.
 
+Currently, there isn't any performant way of detecting whether an `ArrayBuffer` is detached or not. The following implementation is an example of how the detachness can be detected, but has some flaws in v8: Functions with try catch blocks are not inlined in v8. Referencing [Node internal comment](https://github.com/nodejs/node/blob/main/lib/querystring.js#L472).
```
Comment (syg) [inline README.md:114]: Please capitalize v8 to V8.

File: README.md [hunk 2]
```diff
@@ -137,6 +155,12 @@ It is possible to both implement copy-on-write `ArrayBuffer`s and keep the "fixe
 
 At this time, Google Chrome deems this mitigation important enough for security to not implement CoW `ArrayBuffer`s.
 
+### How `transfer` and `get detached()` is used in browsers and in runtimes?
```
Comment (syg) [inline README.md:158]: This seems more accurate as just "How is `get detached()` used in browsers and in runtime?"

**Review by syg — APPROVED** [review]
Thanks! lgtm % comments.

File: README.md (hunk 2 above)
Comment (anonrig) [reply README.md:158]: I'll open another pull request to include the references for `transfer`

**Review by ljharb — COMMENTED** [review]
LGTM, fwiw :-)

**Pull request merged**